### ISSUE TITLE
Claude Testing in browser generates 400 {"type":"error","error": API Request Failed.

### ISSUE THREAD

**Issue by Garry-TI** [cline/cline#675]
### Which API Provider are you using?

Anthropic

### Which Model are you using?

Claude Sonnet 3.5 

### What happened?

Task: **create a prompt engineering services page that mimics @https://geniusee.com/prompt-engineering under our consulting tab in the header of our home page.**
Claude asks to test the changes. and gets to the last test of clicking and moving and gets this error:

400 {"type":"error","error":{"type":"invalid_request_error","message":"messages.0.content.1.image.source.base64.data: At least one of the image dimensions exceed max allowed size for many-image requests: 2000 pixels"}}
![image](https://github.com/user-attachments/assets/06ca0f85-4ac0-4570-a39c-56e841d65a2c)

![image](https://github.com/user-attachments/assets/d02d7fc9-3d43-4fe0-a6c3-5eb802daca06)

### Steps to reproduce

1.gave task
2. approved changes
3. approved browser test

### Relevant API REQUEST output

```shell
400 {"type":"error","error":{"type":"invalid_request_error","message":"messages.0.content.1.image.source.base64.data: At least one of the image dimensions exceed max allowed size for many-image requests: 2000 pixels"}}
in step 6/6
```

**Comment by Garry-TI** [comment]
now I am stuck on this: API Request Failed 400 {"type":"error","error":{"type":"invalid_request_error","message":"messages.132.content.2.image.source.base64.data: At least one of the image dimensions exceed max allowed size for many-image requests: 2000 pixels"}}

![image](https://github.com/user-attachments/assets/7acd12a6-fad1-4bce-ae3a-8ebc246bd526)

I can't seem to get it to not attempt to process the attached images

**Comment by saoudrizwan** [comment]
Once the images are added to the conversation history, they cannot be removed and if it exceeds the image size it bricks the task. I need to add a guard here to throw an error if the user tries to attach an image that exceeds the allowed size. Thank you for bringing this up

**Comment by stuinfla** [comment]
I agree. This is becoming significant issue and keeps causing Cline to error out because I've got 4K screen so when I try to give it information, the images are just too dense and it blows up.

**Comment by jtac** [comment]
I keep running into this same issue when working on my macbook pro, too big screen, clippings are very often over 2000px

suggestion: if someone pastes something above 2000px scale it down, it would be a hassle to have users scale their clippings every time
